python_katas/kata_1/test.py

```diff
@@ -69,78 +111,190 @@ class TestIsUniqueString(unittest.TestCase):
-        # your code here
-        pass
+        test_list = [1, 5, 0, 4, 1, 1, 1]
+        expected = [None, 4, -5, 4, -3, 0, 0]
+        result = questions.list_diff(test_list)
+        self.assertEqual(result, expected)
+
+    def test_empty(self):
+        test_list = []
+        expected = []
+        self.assertEqual(questions.list_diff(test_list), expected)
 
+    def test_one(self):
+        test_list = [1]
+        expected = [None]
+        self.assertEqual(questions.list_diff(test_list), expected)
 
 class TestPrimeNumber(unittest.TestCase):
     """
     1 Katas
     """
 
-    def test_sample(self):
-        # your code here
-        pass
+    def test_prime_1(self):
+        number = 7
+        self.assertEqual(questions.prime_number(number), True)
+
+    def test_prime_2(self):
+        number = 17
+        self.assertEqual(questions.prime_number(number), True)
+
+    def test_prime_3(self):
+        number = 13
+        self.assertEqual(questions.prime_number(number), True)
+
+    def test_not_prime_1(self):
+        number = 1
+        self.assertEqual(questions.prime_number(number), False)
+
+    def test_not_prime_2(self):
+        number = -20
+        self.assertEqual(questions.prime_number(number), False)
+
+    def test_not_prime_3(self):
+        number = 2.4
+        self.assertEqual(questions.prime_number(number), False)
 
 
 class TestPalindromeNum(unittest.TestCase):
     """
     1 Katas
     """
+    def test_pal1True(self):
+        num = 1441
+        self.assertTrue(questions.palindrome_num(num), "the num is palindrome but you return False")
 
-    def test_sample(self):
-        # your code here
-        pass
+    def test_pal1True(self):
+        num = 11
+        self.assertTrue(questions.palindrome_num(num), "the num is palindrome but you return False")
+
+    def test_pal1False(self):
+        num = 113
+        self.assertFalse(questions.palindrome_num(num),"the num is not palindrome but you return True")
+
+    def test_pal1Zero(self):
+        num = 0
+        self.assertTrue(questions.palindrome_num(num)," 0 is palindrom num but you return False")
 
 
 class TestPairMatch(unittest.TestCase):
     """
     3 Katas
     """
 
-    def test_sample(self):
-        # your code here
-        pass
+    def test_Pair1(self):
+        men = {'Ben': 34, 'Ronaldo': 37, 'Ancelotti': 62}
+        women = {'Yasmin': 22, 'Inbar': 18, 'Angelina': 52}
+        self.assertEqual(questions.pair_match(men, women), ('Ancelotti', 'Angelina'))
+
+    def test_Pair2(self):
+        men = {'Roi': 65, 'Eran': 82, 'Ido': 20}
+        women = {'Sivan': 70, 'Orly': 18, 'Neta': 65}
+        self.assertEqual(questions.pair_match(men, women), ('Roi', 'Neta'))
 
 
 class TestBadAverage(unittest.TestCase):
     """
     1 Katas
     """
 
-    def test_sample(self):
-        # your code here
-        pass
+    def easy_peasy(self):
+        nums = [10, 20, 30]
+        self.assertEqual(questions.bad_average(nums), 20)
```

Comment:
The function `bad_average` gets 3 arguments, but you sent single argument which is a list of 3 elements, did you mean `bad_average(10, 20, 30)`?

Reply:
Right, fixed